################################################################################
SHORT DESCRIPTION:
################################################################################
Don't ask qemu if it supports VNC when the config file says vnc=0

################################################################################
LONG DESCRIPTION:
################################################################################
On QMP init, libxl always asks QEMU if it supports VNC, even when disabled.
When QEMU replies by the negative, xl prints an error message in the logs.
This patch adds a check to see if VNC is enabled before probing QEMU.

################################################################################
CHANGELOG
################################################################################
Authors:
Jed Lejosne <lejosnej@ainfosec.com>

################################################################################
REMOVAL
################################################################################
If upstreamed

################################################################################
UPSTREAM PLAN
################################################################################
Seems like a good candidate for upstream

################################################################################
INTERNAL DEPENDENCIES
################################################################################

################################################################################
PATCHES
################################################################################
--- a/tools/libxl/libxl_qmp.c
+++ b/tools/libxl/libxl_qmp.c
@@ -1296,7 +1296,7 @@ int libxl__qmp_initializations(libxl__gc
         ret = qmp_change(gc, qmp, "vnc", "password", vnc->passwd);
         qmp_write_domain_console_item(gc, domid, "vnc-pass", vnc->passwd);
     }
-    if (!ret) {
+    if (!ret && vnc) {
         ret = qmp_query_vnc(qmp);
     }
     libxl__qmp_close(qmp);
